src/common/list-syntax.k

```diff
@@ -0,0 +1,309 @@
+syntax TypeParamList ::= List{TypeParam,","} [klabel('TypeParamList)]
+syntax AbstractMethodModList ::= List{AbstractMethodMod, ""} [klabel('AbstractMethodModList)]
+syntax AnnoAbstractMethodModList ::= List{AnnoAbstractMethodMod, ""} [klabel('AnnoAbstractMethodModList)]
+syntax AnnoMethodModList ::= List{AnnoMethodMod, ""} [klabel('AnnoMethodModList)]
+syntax AnnoVarModList ::= List{AnnoVarMod, ""} [klabel('AnnoVarModList)]
+syntax AnnoClassModList ::= List{AnnoClassMod, ""} [klabel('AnnoClassModList)]
+syntax AnnoConstrModList ::= List{AnnoConstrMod, ""} [klabel('AnnoConstrModList)]
+syntax AnnoConstantModList ::= List{AnnoConstantMod, ""} [klabel('AnnoConstantModList)]
+syntax AnnoFieldModList ::= List{AnnoFieldMod, ""} [klabel('AnnoFieldModList)]
+syntax AnnoInterfaceModList ::= List{AnnoInterfaceMod, ""} [klabel('AnnoInterfaceModList)]
+syntax AnnoList ::= List{Anno, ""} [klabel('AnnoList)]
+syntax AnnoElemDecList ::= List{AnnoElemDec, ""} [klabel('AnnoElemDecList)]
+syntax InterfaceMemberDecList ::= List{InterfaceMemberDec, ""} [klabel('InterfaceMemberDecList)]
+syntax ElemValPairList ::= List{ElemValPair,","} [klabel('ElemValPairList)]
+syntax ElemValList ::= List{ElemVal,","} [klabel('ElemValList)]
+syntax StringPartList ::= List{StringPart,""} [klabel('StringPartList)]
+syntax EnumConstList ::= List{ EnumConst, ","} [klabel(' EnumConstList)]
+syntax ClassBodyDecList ::= List{ClassBodyDec, ""} [klabel('ClassBodyDecList)]
+syntax BlockStmList ::= List{BlockStmt, ""} [klabel('BlockStmList)]
+syntax SwitchGroupList ::= List{SwitchGroup, ""} [klabel('SwitchGroupList)]
+syntax SwitchLabelList ::= List{SwitchLabel, ""} [klabel('SwitchLabelList)]
+
+
+syntax UserList ::=     BlockStmList
+                      | SwitchGroupList
+                      | SwitchLabelList
+                      | InterfaceTypeList
+                      | ExceptionTypeList
+                      | IdList
+                      | TypeDecList
+                      | VarDecList
+                      | ImportDecList
+                      | ActualTypeArgList
+                      | ClassOrInterfaceTypeList
+                      | TypeParamList
+                      | AbstractMethodModList
+                      | AnnoAbstractMethodModList
+                      | AnnoMethodModList
+                      | AnnoVarModList
+                      | AnnoClassModList
+                      | AnnoConstrModList
+                      | AnnoConstantModList
+                      | AnnoFieldModList
+                      | AnnoInterfaceModList
+                      | AnnoList
+                      | AnnoElemDecList
+                      | InterfaceMemberDecList
+                      | ElemValPairList
+                      | ElemValList
+                      | StringPartList
+                      | EnumConstList
+                      | ClassBodyDecList
+
+/* Syntax below should be put in other modules when they are perfectly defined. */
```

Comment:
Here you should use ```TODO: Syntax below ...```, because later you can search after ```TODO```, and also some editors recognize it.